File: AutomaticTimetableGeneratorSystem-main/timetable-frontend/src/services/facultyService.js
import api from "./api";

const BASE_URL = "/faculties";

const facultyService = {

    // Get All Faculty
    getAllFaculty: async () => {

        const response = await api.get(BASE_URL);

        return response.data;

    },

    // Get Faculty By Id
    getFacultyById: async (id) => {

        const response = await api.get(`${BASE_URL}/${id}`);

        return response.data;

    },

    // Create Faculty
    createFaculty: async (faculty) => {

        const response = await api.post(
            BASE_URL,
            faculty
        );

        return response.data;

    },

    // Update Faculty
    updateFaculty: async (id, faculty) => {

        const response = await api.put(
            `${BASE_URL}/${id}`,
            faculty
        );

        return response.data;

    },

    // Delete Faculty
    deleteFaculty: async (id) => {

        const response = await api.delete(
            `${BASE_URL}/${id}`
        );

        return response.data;

    }

};

export default facultyService;

File: AutomaticTimetableGeneratorSystem-main/timetable-backend/src/main/java/com/timetable/service/FacultyService.java
package com.timetable.service;

import java.util.List;

import com.timetable.dto.faculty.FacultyRequest;
import com.timetable.dto.faculty.FacultyResponse;

public interface FacultyService {

    /**
     * Save Faculty
     */
    FacultyResponse saveFaculty(FacultyRequest request);

    /**
     * Update Faculty
     */
    FacultyResponse updateFaculty(Long facultyId,
                                  FacultyRequest request);

    /**
     * Delete Faculty
     */
    void deleteFaculty(Long facultyId);

    /**
     * Get Faculty By ID
     */
    FacultyResponse getFacultyById(Long facultyId);

    /**
     * Get All Faculties
     */
    List<FacultyResponse> getAllFaculties();

}